tests/parametric/test_otel_span_methods.py

```diff
@@ -348,3 +377,437 @@ def test_otel_get_span_context(self, test_agent, test_library):
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    @pytest.mark.parametrize(
+        "span_kind,expected_name", [(SK_INTERNAL, "internal"), (SK_CONSUMER, "consumer"), (SK_PRODUCER, "producer")]
+    )
+    def test_otel_span_operation_name_span_kind(self, span_kind: int, expected_name: str, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `span.kind` (barring `client` and `server`) (in this case "internal") when:
+            - Span kind is set to Internal
+            - no other known attributes for setting the operation name
+        """
+        run_operation_name_test(
+            name="otel_span_name",
+            expected_name=expected_name,
+            span_kind=span_kind,
+            attributes=None,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_unknown(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to "otel_unknown" when:
+            - span.kind is not present
+            - no other known attributes for setting the operation name
+        """
+        run_operation_name_test(
+            name="otel_span_name",
+            expected_name="otel_unknown",
+            span_kind=None,
+            attributes=None,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_special_tag_overrides(self, test_agent, test_library):
+        """
+            Tests that the special tags will override expected values
+        """
```

Comment:
Let's call these "reserved attributes" instead, just to make this very clear what they are. (This came from a discussion with my team earlier today)
I'll update the docs and tickets accordingly

Reply:
Done in 5d3887ccdb9ab5d9ace73786b4df961bbbf92679